src/lib/input/input-wrapper.ts

```diff
@@ -0,0 +1,289 @@
+import {
+  Component,
+  Input,
+  Directive,
+  AfterContentInit,
+  ContentChild,
+  SimpleChange,
+  ContentChildren,
+  ElementRef,
+  QueryList,
+  OnChanges,
+  ViewEncapsulation
+} from '@angular/core';
+import {MdError, coerceBooleanProperty} from '../core';
+import {NgModel} from '@angular/forms';
+
+
+// Invalid input type. Using one of these will throw an MdInputWrapperUnsupportedTypeError.
+const MD_INPUT_INVALID_INPUT_TYPE = [
+  'file',
+  'radio',
+  'checkbox',
+];
+
+
+let nextUniqueId = 0;
+
+
+export class MdInputWrapperInputElementCountError extends MdError {
+  constructor(count: number) {
+    super(`md-input-wrapper must contain exactly 1 input or textarea element. Found ${count}.`);
+  }
+}
+
+export class MdInputWrapperPlaceholderConflictError extends MdError {
```

Comment:
I don't see the harm in keeping them separate for now, especially since we're just going to delete the md-input version shortly. It would be weird for MdInput to throw MdInputWrapper* errors